--- a/drivers/spi/spi-fsl-espi.c
+++ b/drivers/spi/spi-fsl-espi.c
@@ -395,12 +395,14 @@ static void fsl_espi_rw_trans(struct spi
 
 		espi_trans->n_tx = n_tx;
 		espi_trans->n_rx = trans_len;
-		espi_trans->len = trans_len + n_tx;
+		//espi_trans->len = trans_len + n_tx;
+		espi_trans->len = n_tx;
 		espi_trans->tx_buf = local_buf;
 		espi_trans->rx_buf = local_buf + n_tx;
 		fsl_espi_do_trans(m, espi_trans);
 
-		memcpy(rx_buf + pos, espi_trans->rx_buf + n_tx, trans_len);
+		//memcpy(rx_buf + pos, espi_trans->rx_buf + n_tx, trans_len);
+		memcpy(rx_buf + pos, espi_trans->rx_buf, trans_len);
 
 		if (loop > 0)
 			espi_trans->actual_length += espi_trans->len - n_tx;
@@ -429,6 +431,7 @@ static void fsl_espi_do_one_msg(struct s
 	espi_trans.n_tx = n_tx;
 	espi_trans.n_rx = n_rx;
 	espi_trans.len = n_tx + n_rx;
+	//espi_trans.len = n_tx;
 	espi_trans.actual_length = 0;
 	espi_trans.status = 0;
 
@@ -692,6 +695,7 @@ static int of_fsl_espi_probe(struct plat
 	struct resource mem;
 	struct resource irq;
 	int ret = -ENOMEM;
+	printk("fsl-espi starting...");
 
 	ret = of_mpc8xxx_spi_probe(ofdev);
 	if (ret)
